backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/transformer/DecisionTransformer.java

```diff
@@ -673,6 +695,26 @@ public static DocumentationUnit transformToDomain(DecisionDTO decisionDTO) {
             .creatingDocOffice(
                 DocumentationOfficeTransformer.transformToDomain(
                     decisionDTO.getCreatingDocumentationOffice()))
+            .source(
+                decisionDTO.getSource().stream()
+                    .max(Comparator.comparing(SourceDTO::getRank)) // Find the highest-ranked item
+                    .map(
+                        sourceDTO -> {
+                          SourceValue sourceValue = null;
+                          if (sourceDTO.getValue() != null) {
+                            try {
+                              sourceValue = sourceDTO.getValue();
+                            } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
Kann es hier überhaupt zu einer NPE kommen? Wir sind ja eigentlich schon in dem entsprechenden if()-block

Reply:
Ja guter Punkt! Hatte das auch schon entfernt :)